crates/gpui/src/window/element_cx.rs

```diff
@@ -758,6 +758,38 @@ impl<'a> ElementContext<'a> {
         );
     }
 
+    /// Paint a strikethrough into the scene for the next frame at the current z-index.
+    pub fn paint_strikethrough(
+        &mut self,
+        origin: Point<Pixels>,
+        width: Pixels,
+        style: &StrikethroughStyle,
+    ) {
+        let scale_factor = self.scale_factor();
+        let height = style.thickness;
+        let bounds = Bounds {
+            origin,
+            size: size(width, height),
+        };
+        let content_mask = self.content_mask();
+        let view_id = self.parent_view_id();
+
+        let window = &mut *self.window;
+        window.next_frame.scene.insert(
+            &window.next_frame.z_index_stack,
+            Underline {
```

Comment:
Does it make sense to have a separate rendering primitive for strikethroughs or should this be given a generic name (HorizontalLine)?

Reply:
Totally up to you at this point, I guess.
I would not use a generic name in the beginning at least, and since we use it only once here, maybe fine to leave it be for now?